### PR TITLE
Changed user schema, added OAuth login, create, current, logout and l…

### PR DESCRIPTION
**Pull request by MiraMarshall**
Added OAuth controller methods login, logout, create, and current. Changed schema to match GitHub auth hash. Will need to run rails db:migrate after pull and change other tests to username instead of name.

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -13,6 +13,63 @@ def show
     end
   end
 
+  def login_form
```
Comment (alexandria7) [inline app/controllers/users_controller.rb]: do we actually need login_form or login methods anymore after we have the create method that uses OmniAuth?

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -13,6 +13,63 @@ def show
...
+    @user = User.new
+  end
+
+  def login
```
Comment (alexandria7) [inline app/controllers/users_controller.rb:16]: same as above - I thought github now does this for us

File: test/controllers/users_controller_test.rb [hunk 3]
```diff
@@ -32,4 +32,41 @@
       expect(flash[:result_text]).must_equal "User not found!"
     end
   end
+
+  describe "login" do
```
Comment (alexandria7) [inline test/controllers/users_controller_test.rb:36]: this is testing your "create" action not your login action - maybe either change the name of this test?

File: test/controllers/users_controller_test.rb [hunk 4]
```diff
@@ -32,4 +32,41 @@
...
+    it "can log in an existing user" do
+      # Arrange
+      user_count = User.count
+
+      # Act
+      user = perform_login
+
+      expect(user_count).must_equal User.count
```
Comment (alexandria7) [inline test/controllers/users_controller_test.rb]: is this actually testing anything? or rather, i'm not sure what this is testing here

Comment (alexandria7) [reply test/controllers/users_controller_test.rb]: i might have an expect here, like: expect { user = perform_login(user) }.wont_change "User.count"

File: test/controllers/users_controller_test.rb [hunk 5]
```diff
@@ -32,4 +32,41 @@
...
+
+      # Should also test Flash notices
```
Comment (alexandria7) [inline test/controllers/users_controller_test.rb]: absolutely agree! i might have a test that says: expect(flash[:success]).must_equal "Logged in as returning user #{user.username}"

File: test/controllers/users_controller_test.rb [hunk 6]
```diff
@@ -32,4 +32,41 @@
...
+    it "can log in a new user" do
+      # Arrange
+      user = User.new(provider: "github", username: "peter", uid: 987, email: "peter@griffin.com")
+
+      expect {
+        # Act
+        perform_login(user)
+        # Assert
+      }.must_change "User.count", 1
+
```
Comment (alexandria7) [inline test/controllers/users_controller_test.rb:57]: nice expect!

File: test/controllers/users_controller_test.rb [hunk 7]
```diff
@@ -32,4 +32,41 @@
...
+      # Should also test Flash notices
```
Comment (alexandria7) [inline test/controllers/users_controller_test.rb]: for sure! i'd throw in expect(flash[:success]).must_equal "Logged in as new user #{user.username}"

File: test/controllers/users_controller_test.rb [hunk 8]
```diff
@@ -32,4 +32,41 @@
...
+      user = User.find_by(uid: user.uid, provider: user.provider)
+
+      expect(session[:user_id]).must_equal user.id
+    end
+  end
+
```
Comment (alexandria7) [inline test/controllers/users_controller_test.rb:64]: we should also test to make sure that if the user provides invalid info (like no name or whatever is required), we get the error messages ("Could not create new user account: #{user.errors.messages}") we are expecting and a redirect

**Review by alexandria7 — CHANGES_REQUESTED** [review]
<3

File: app/controllers/users_controller.rb [hunk 9]
```diff
@@ -13,6 +13,59 @@ def show
...
+
+  def current
+    @user = User.find_by(id: session[:user_id])
+    if @user.nil?
+      flash[:error] = "You must be logged in first!"
+      redirect_to users_path
+    end
+  end
+
+  def logout
```
Comment (AngelaOh) [inline app/controllers/users_controller.rb:45]: These look great!! I think the logout is defined as "destroy" in our route so we can change one or the other to match each other!

**Pull request merged**